### ISSUE TITLE
Assertion failed: Assertion failed on expression: 'verts.size() > 0'

### ISSUE THREAD

**Issue by sinistersnare** [Unity-Technologies/NavMeshComponents#85]
Hi, I am using this to build a NavMesh at run-time. It works great, thanks so much! Sometimes, however, I get an error (the title of this issue). It does not seem to affect the game at all, so I wonder if it is something that I am doing. Here is the stack trace:

```txt
Assertion failed: Assertion failed on expression: 'verts.size() > 0'
UnityEngine.AI.NavMeshSurface:BuildNavMesh()
LevelCreator:CreateNavMesh() (at Assets/Scripts/procgen/LevelCreator.cs:211)
LevelCreator:Start() (at Assets/Scripts/procgen/LevelCreator.cs:75)
```

line 221 of LevelCreator is simple

```csharp
surface.BuildNavMesh();
```

so I think its at your code, but I do not see any assertions in `BuildNavMesh`. Hope I can help fix this!

**Comment by turadr** [comment]
Hi, This issue has now been resolved in Unity 2018.3.0a10 . We'll soon solve it in other editor versions as well.

https://issuetracker.unity3d.com/issues/assertion-failed-on-expression-verts-dot-size-0-is-shown-when-navmeshobstacle-has-carve-enabled

**Comment by monsterblues** [comment]
Looks like it was patched in 2017.4.14f1 as well.
